main/webapp/modules/core/scripts/index/default-importing-controller/controller.js

```diff
@@ -265,10 +265,13 @@ Refine.DefaultImportingController.prototype.getPreviewData = function(callback,
           "limit" : numRows || 100 // More than we parse for preview anyway
         }),
         null,
-        function(data) {
-          result.rowModel = data;
-          callback(result);
-        },
+		  function(data) {
+			  if (data["rows"].length == 0) {
+				alert($.i18n('core-index-import/load-xml-rows-error'));
+			  }
+			  result.rowModel = data;
+			  callback(result);
+		  },
         "jsonp"
```

Comment:
@WaltonG: I understand that it was not there before, but how about putting some error handling here? It seams to me that if the call fails, the user is going to be stuck...

Reply:
@antoine2711 Actuallythe callback was there before.  I only added
https://github.com/OpenRefine/OpenRefine/blob/50e84ebb43bee103113540a47191151856e319f3/main/webapp/modules/core/scripts/index/default-importing-controller/controller.js#L269-L271

Reply:
> @WaltonG: I understand that it was not there before, but how about putting some error handling here? It seams to me that if the call fails, the user is going to be stuck...

@antoine2711: I cant seem to get the errors that will make the user get stuck. I would appreciate some pointers to proceed

Reply:
> @antoine2711: I cant seem to get the errors that will make the user get stuck. I would appreciate some pointers to proceed

Network error. It's an API call, no?